<launch>   
  <arg name="use_last_pose" default="true"/>	
  <arg name="map_file" default="$(find squirrel_navigation)/maps/default-map.yaml" />
  <arg name="localizer_no_map" default="false"/>
  
  <!-- Run map_server -->
  <include file="$(find squirrel_navigation)/launch/map_server.launch" unless="$(arg localizer_no_map)">
    <arg name="map_file" value="$(arg map_file)" />
  </include>

  <!-- Run scan matcher -->
  <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" name="laser_scan_matcher_node" output="screen">
    <param name="base_frame" value="/base_link"/>
    <param name="fixed_frame" value="/odom"/>
    <param name="max_iterations" value="10"/>
  </node>

  <!-- Pose Transformer -->
  <node name="scan_odom_tf_pub" pkg="squirrel_localizer" type="squirrel_localizer_tf_pose_node" output="screen">
    <param name="pose_topic" value="squirrel_localizer_odom_pose"/>
    <param name="from_id" value="map"/>
    <param name="to_id" value="odom"/>
  </node>

  <!-- Localiser -->
  <node name="localizer" pkg="squirrel_localizer" type="squirrel_localizer_localization_node" output="screen">
    <param name="verbose" value="false"/>
    <param name="number_of_particles" value="5000"/>
    <param name="maxLocalizationRange" value="5.0"/>
    <param name="maxRange" value="6.0"/>
    <param name="motion_param_magnitude" value="2.5"/>
    <param name="angular_update" value="0.05"/>
    <param name="linear_update" value="0.1"/>
    <param name="motion_param_ff" value="0.2"/>
    <param name="motion_param_fs" value="0.1"/>
    <param name="motion_param_fr" value="0.2"/>
    <param name="motion_param_rr" value="0.2"/>
    <param name="motion_param_sth" value="1.0"/>
    <param name="motion_param_sx" value="1.0"/>
    <param name="motion_param_sy" value="1.0"/>
    <param name="observationSigma" value="0.6"/>
    <param name="use_second_laser" value="false"/>
    <param name="base_link_frame_ID" value="base_link"/>
    <param name="odom_frame_ID" value="odom"/>
    <param name="front_laser_frame_ID" value="hokuyo_link"/>
    <param name="scan_front_topic" value="/scan"/>
    <param name="rear_laser_frame_ID" value="hokuyo_link"/>
    <param name="scan_rear_topic" value="/scan"/>
    <param name="sm_point2line_outliers_maxPerc"  value="0.5"/>
    <param name="sm_use_trimed_mean_as_reference_scan"  value="false"/>
    <param name="dMapTheshold" value="0.5"/>
    <param name="observationPointDensity" value="0.5"/>
    <param name="use_last_pose" value="$(arg use_last_pose)"/>
    <param name="use_scan_to_map_matching" value="false"/>
    <param name="sm_filter_gain" value="0.825"/>
  </node>

</launch>
